Path deform tool update.

seagetch opened the pull request:
Added bunch of changes to path deform tool.

1. Partial deformation for selected vertices.
     Path deformation tool can select vertices when pressing Alt key. When applying deformation, only selected vertices are moved.

2. Update undo / redo of path deformation tool.
     Redo / undo of Path deformation tool remembers the keypoint of the mesh editor.

3. Partial reset / flipping
     When selecting vertices, right click menu shows "Reset selected" and "Flip selected horizontally from mirror" on the menu. It applies the reset / flip of the deformation only to the selected vertices.

seagetch committed:
Adding tool option buttons for path deformation tool.

seagetch committed:
Implements vertex selection and partial deform for path deformation tool.

seagetch committed:
WIP: Improve redo / undo of path deformation tool (still have some problematic situation.)

seagetch committed:
Implement partial reset / flipping of selected drawables.

seagetch committed:
Merge branch 'v0_8' of https://github.com/Inochi2D/inochi-creator into deform_ui

LunaTheFoxgirl commented on source/creator/utils/transform.d:
```diff
@@ -0,0 +1,149 @@
+    *   axis = 0: flip horizontally, 1: flip vertically, -1: flip diagonally, 2: not flipped
+    *   extrapolation = specifying source index is selected in mirroered position or not.
+    */
+void autoFlipBinding(ParameterBinding binding, ParameterBinding srcBinding, vec2u index, uint axis, bool extrapolation = true, ulong[]* selected = null) {
```
Freestanding functions should use the inc\<Space\>\<Function Name\> convention.

In this case incBindingAutoFlip

seagetch replied:
renamed autoFlipBinding to incBindingAutoFlip.

LunaTheFoxgirl commented on source/creator/utils/transform.d:
```diff
@@ -0,0 +1,149 @@
+
+
+
+ParameterBinding getPairBindingFor(Parameter param, Node target, FlipPair pair, string name, bool forceCreate = false) {
```
See autoFlipBinding,

incBindingGetPairFor, would be appropriate here

seagetch replied:
renamed getPairBindingFor to incBindingGetPairFor.

LunaTheFoxgirl reviewed (changes requested):
We should keep the code style consistent, otherwise this commit is almost ready to merge.

seagetch committed:
Renamed some common utility functions.

- autoFlipBinding to incBindingAutoFlip
- getPairBindingFor to incBindingGetPairFor

seagetch committed:
Fix comment for incBindingAutoFlip.

LunaTheFoxgirl reviewed (approved):
LGTM, merging
